activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -412,6 +412,7 @@ def include?(record)
     # argument.
     def raise_record_not_found_exception!(ids = nil, result_size = nil, expected_size = nil, key = primary_key, not_found_ids = nil) # :nodoc:
       conditions = " [#{arel.where_sql(klass)}]" unless where_clause.empty?
+      ids ||= where_clause.to_h[key]
```

Comment:
It's a nitpick but I don't think this accounts for models with a composite primary key